### PR TITLE
feat: Rename role

### PR DESCRIPTION
**Pull request by 00LT00**
#613

### CODE REVIEW

**Comment by 00LT00** [comment]
plz review @hsluoyz @nodece @GopherJ

File: management_api_test.go [hunk 1]
```diff
@@ -253,4 +253,12 @@ func TestModifyGroupingPolicyAPI(t *testing.T) {
 	testGetUsers(t, e, []string{}, "data1_admin")
 	testGetUsers(t, e, []string{}, "data2_admin")
 	testGetUsers(t, e, []string{"eve"}, "data3_admin")
+	_, _ = e.AddGroupingPolicy("alice", "eve")
```
Comment (hsluoyz) [inline management_api_test.go:256]: @00LT00 `eve` is usually a user name, not a role name.

File: management_api.go [hunk 2]
```diff
@@ -296,6 +296,33 @@ func (e *Enforcer) RemoveNamedGroupingPolicies(ptype string, rules [][]string) (
...
+	roles, err := e.GetRolesForUser(oldName)
+	if err != nil {
+		return false, err
+	}
+	names, err := e.GetUsersForRole(oldName)
+	if err != nil {
+		return false, err
+	}
+	for _, role := range roles {
+		if flag, err := e.updatePolicy("g", ptype, []string{oldName, role}, []string{newName, role}); err != nil || !flag {
```
Comment (GopherJ) [inline management_api.go:313]: it doesn't seems correct to me because here we are assuming: `g` only have two parameters but it's not always this case

Comment (00LT00) [reply management_api.go:313]: Do you mean we should add UpdateRoleInDomain? 
Or is it like UpdatePolicy?
```
func (e *Enforcer) UpdateNamedPolicy(ptype string, p1 []string, p2 []string) (bool, error) {
	return e.updatePolicy("p", ptype, p1, p2)
}
```

Comment (00LT00) [reply management_api.go]: I think the new logic could make sure this problem.

**Review by nodece — APPROVED** [review]
LGTM

**Comment by nodece** [comment]
@00LT00 your commit message looks confusing.

File: management_api.go (hunk 2 above)
Comment (hsluoyz) [reply management_api.go]: Any update?

**Pull request merged**